iquip/apps/scheduler.py

```diff
@@ -0,0 +1,307 @@
+)
+
+import qiwis
+from iquip.protocols import ExperimentInfo
+
+def delete_items_of_layout(layout: QLayout):
+    """Auxiliary function for deleting components in the layout.
+
+    Args:
+        layout: The layout whose elements are intended to be deleted.
+    """
+    if layout is not None:
+        while layout.count():
+            item = layout.takeAt(0)
+            widget = item.widget()
+            if widget is not None:
+                widget.setParent(None)
+            else:
+                delete_items_of_layout(item.layout())
+
+
+class SchedulerFrame(QWidget):
+    """Frame for displaying the submitted experiment list.
+    
+    Attributes:
+        expRun: The ExperimentView widget displaying currently running experiment.
+        expList: The QListView widget holding queued experiments.
+        model: The ExperimentModel that manages the data of expList.
+    """
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended. """
+        super().__init__(parent=parent)
+        layout = QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 20)
+        self.expRun = RunningExperimentView(self)
+        self.expList = QListView(self)
+        self.expList.setItemDelegate(ExperimentDelegate(self.expList))
+        self.expList.setMovement(QListView.Free)
+        self.expList.setObjectName("Queued Experiments")
+        self.model = ExperimentModel([])
+        self.expList.setModel(self.model)
+        layout.addWidget(QLabel("Currently running:"))
+        layout.addWidget(self.expRun)
+        layout.addWidget(QLabel("Queued experiments:"))
+        layout.addWidget(self.expList)
+        self.setLayout(layout)
+
+class ExperimentModel(QAbstractListModel):
+    """Model for managing the data in the experiment list.
+    
+    Attributes:
+        data: The list of ExperimentView widget.
+    """
+
+    def __init__(self, data: list, parent=None):
+        super().__init__(parent)
+        self.expData = data
+
+    def rowCount(self, parent=QModelIndex()):
+        """Overrided."""
+        return len(self.expData)
+
+    def data(self, index: QModelIndex, role=Qt.DisplayRole):
+        """Overrided."""
+        return self.expData[index.row()]
+
+    def supportedDropActions(self):
+        """Overrided."""
+        return Qt.CopyAction | Qt.MoveAction
+
+    def mimeData(self, index: QModelIndex) -> QMimeData:
+        """Fetches the index of the selected element.
+        
+        Args:
+            index: The ModelIndex instance containing 
+                   row, column, and etc. values of the selected element.
+        """
+        mime = super().mimeData(index)
+        mime.setText(str(index[0].row()))
+        return mime
+
+    def dropMimeData(self, # pylint: disable=too-many-arguments
+        mimedata: QMimeData,
+        action: QAction,
+        row: int,
+        column: int,
+        parentIndex: QModelIndex):
```

Comment:
If there is a return value, breaking the last line would be better:
```python
def f(
    a: A,
    b: B,
) -> C:  # pylint: ...
    ...
```
Moreover, please write the pylint comment after the definition (after the colon).
Note that there are 2 spaces before `#`.

Reply:
In addition, if a function/method returns something, it should describe the return value in its docstring. Please read [Google Python style guide](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods).